components/camera/camera.go

```diff
@@ -470,19 +492,32 @@ func (c *reconfigurableCamera) DoCommand(ctx context.Context, cmd map[string]int
 }
 
 // Reconfigure reconfigures the resource.
-func (c *reconfigurableCamera) Reconfigure(_ context.Context, newCamera resource.Reconfigurable) error {
+func (c *reconfigurableCamera) Reconfigure(ctx context.Context, newCamera resource.Reconfigurable) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	actual, ok := newCamera.(*reconfigurableCamera)
 	if !ok {
 		return utils.NewUnexpectedTypeError(c, newCamera)
 	}
+	if err := viamutils.TryClose(ctx, c.actual); err != nil {
```

Comment:
I'm going to venture a guess that we removed the TryClose here because of the double reconfigure logic. now that it is refactored and has a notifier pattern, I think it is critical we bring the close back. Otherwise we leak resources over time.

Reply:
The reason we removed this didn't have anything to do with the double reconfigure logic. It was this comment [here](https://github.com/viamrobotics/rdk/pull/1644/files#r1053882599). In short, both the new and old camera will point to the same file in Linux. If you create a new camera that points to the same underlying file as the old camera and close the old camera, the old camera will close the file that the new camera points to and the new camera will not work. You'll get `EOF` errors trying to read from the new camera.

It looks like that error is being brought back.

Reply:
but we're not doing that now because the notify system just replaces the camera internally, not through this method. something created outside of this should probably have the old close.

we still don't cover the case well of the same physical camera being reconfigured to say a different resolution. this requires us properly handling UpdateAction and Reconfigure past the Reconfigurable, which no code is doing.

I have seen no error happening with reconnecting due to the new logic. I think we should keep this code here and file a new ticket for handling reconfiguration (parameter change).

Reply:
> I have seen no error happening with reconnecting due to the new logic. 

Well we aren't calling this method anymore in `reconnectCamera` (previously `reconfigureCamera`) so we wouldn't see it here but we would see it whenever this method is used. The error never surfaced until I tried to actually use the function to reconfigure the camera.

> but we're not doing that now because the notify system just replaces the camera internally, not through this method. something created outside of this should probably have the old close.

We're doing that in _this_ code but if anyone uses this function without calling close on the camera _before_ creating the new camera it will break the new camera. If the caller has already closed the old camera then this wouldn't be needed.